### PR TITLE
allocate less in Encode

### PR DESCRIPTION
**Pull request by josharian**

### CODE REVIEW

**Comment by onyb** [comment]
@josharian Needs a rebase following #183.

**Comment by josharian** [comment]
If I take the time to rebase will someone review and merge? It has been sitting unreviewed for two months.

**Comment by jcvernaleo** [comment]
Hi @josharian

Sorry for letting this one sit.  We are getting better at prioritizing and not missing PRs but aren't perfect at it yet so sometimes ones get missed.  As our backlog gets smaller I'm hoping for it to happen less though.  At least that is my hope :)

Anyway, yes, please rebase and I'll make sure we review it.  Will add it to our project board to make sure we don't miss it again.

**Comment by josharian** [comment]
Done. #183 picked up the lowest hanging fruit that this PR addressed, but there was a still a small performance benefit available.

File: base58/base58.go [hunk 1]
```diff
@@ -90,9 +89,11 @@ func Encode(b []byte) string {
 	x := new(big.Int)
 	x.SetBytes(b)
 
-	answer := make([]byte, 0, len(b)*136/100)
+	// maximum length of output is log58(2^(8*len(b))) == len(b) * 8 / log(58)
+	maxlen := int(float64(len(b))*1.365658237309761) + 1
```
Comment (onyb) [inline base58/base58.go:93]: You could probably get away with a `float32` here, which is still significantly bigger than what `len(b)` could ever be. Thoughts?

**Review by onyb — APPROVED** [review]
LGTM! Made a small comment (not that important).

File: base58/base58.go (hunk 1 above)
Comment (josharian) [reply base58/base58.go:93]: My 2c: Doesn't really matter. There'll never be a detectable performance impact, it's equally readable either way. Not worth changing.

**Review by jcvernaleo — APPROVED** [review]
Looks good.
OK

**Pull request merged**